cmake/HPX_Libraries.cmake

```diff
@@ -25,11 +25,35 @@ macro(hpx_libraries)
     if(NOT _skip)
       list(FIND HPX_LIBRARIES "${lib}" _found)
       if(_found EQUAL -1)
-        set(_libs ${_libs} ${_keyword} ${lib})
+        if(IS_ABSOLUTE ${lib})
+          hpx_create_imported_library(${lib} _target)
+          set(_libs ${_libs} ${_keyword} ${_target})
+        else()
+          set(_libs ${_libs} ${_keyword} ${lib})
+        endif()
       endif()
       set(_keyword "")
       set(_skip FALSE)
     endif()
   endforeach()
   set(HPX_LIBRARIES ${HPX_LIBRARIES} ${_libs} CACHE INTERNAL "" FORCE)
 endmacro()
+
+macro(hpx_create_imported_library path target)
+  get_filename_component(_basename ${path} NAME_WE)
+  set(${target} hpx::imported::${_basename})
+
+  if(NOT TARGET ${${target}})
+    get_filename_component(_extension ${path} EXT)
+    if(_extension STREQUAL ".so")
+      set(_link_type SHARED)
+    elseif(_extension STREQUAL ".a")
+      set(_link_type STATIC)
+    else()
+      message(FATAL_ERROR "Unknown library extension (${_extension}): ${path}")
```

Comment:
I agree. I should revert this back to how I had it originally with UNKNOWN link type since I don't think it breaks any more tests than the tests that don't already compile.